### PR TITLE
table-filter: add column filter

### PR DESCRIPTION
**Pull request by 0xPoe**
### What problem does this PR solve? 

ref: https://github.com/pingcap/ticdc/pull/3030

part of https://github.com/pingcap/ticdc/issues/3082

### What is changed and how it works?

add column filter. The rules are the same as table filter except that there is no separator.
### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

None

Related changes
None

### CODE REVIEW

**Comment by 0xPoe** [comment]
/cc @kennytm @lichunzhu @overvenus 

I don't know if the current extension of parse makes sense, maybe there is a better way to extend and abstract it. Please feel free to make suggestions! Thanks!

**Comment by 0xPoe** [comment]
/hold

Please do not merge it until TiCDC accepts the RFC for this feature. I will submit the RFC to TiCDC tomorrow.

File: pkg/table-filter/column_filter.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filter
+
+import "strings"
+
+// ColumnFilter is a structure to check if a column should be included for processing.
+type ColumnFilter interface {
```
Comment (3AceShowHand) [inline pkg/table-filter/column_filter.go:19]: how about add a interface, named `Matcher`, then implement a concret type named `ColumnSelector`

Comment (0xPoe) [reply pkg/table-filter/column_filter.go:19]: This is already an interface, and columnFilter is its implementation.

File: pkg/table-filter/helper.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+package filter
+
+import "sort"
+
+// reversing replace the contents of a slice with the same elements but in reverse order.
+func reversing(items interface{}) {
+	sort.SliceStable(items, func(i, j int) bool {
+		return true
+	})
+}
```
Comment (kennytm) [inline pkg/table-filter/helper.go:26]: wut, is this an O(n log n) reverse method 😭

Comment (0xPoe) [reply pkg/table-filter/helper.go:26]: Fixed! Thanks for your review!

File: pkg/table-filter/column_filter.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+	return columnFilter(p.rules), nil
+}
+
+// MatchColumn checks if a column can be processed after applying the columnFilter `f`.
+// Column is not case-sensitive on any platform, nor are column aliases.
+// So we always match in lowercase.
+func (f columnFilter) MatchColumn(column string) bool {
+	for _, rule := range f {
+		if rule.column.matchString(strings.ToLower(column)) {
```
Comment (kennytm) [inline pkg/table-filter/column_filter.go]: move the computation of `strings.ToLower(column)` out of the loop

**Review by kennytm — COMMENTED** [review]
rest lgtm.

is the "cdc rfc" submitted?

File: pkg/table-filter/column_filter.go (hunk 3 above)
Comment (0xPoe) [reply pkg/table-filter/column_filter.go]: Fixed. Good catch! Thanks!

**Comment by 0xPoe** [comment]
> is the "cdc rfc" submitted?

See: https://github.com/pingcap/ticdc/pull/3030

File: pkg/table-filter/column_filter_test.go [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+		_, err := filter.ParseColumnFilter([]string{tc.arg})
+		c.Assert(err, ErrorMatches, tc.msg, Commentf("test case = %s", tc.arg))
+	}
+}
+
+func (s *columnFilterSuite) TestImport(c *C) {
+	dir := c.MkDir()
+	path1 := filepath.Join(dir, "1.txt")
+	path2 := filepath.Join(dir, "2.txt")
+	ioutil.WriteFile(path1, []byte(`
```
Comment (sleepymole) [inline pkg/table-filter/column_filter_test.go]: Nit: `ioutil.WriteFile` can be replaced by `os.WriteFile`.

Comment (0xPoe) [reply pkg/table-filter/column_filter_test.go]: Addressed! Thanks for your review!

File: pkg/table-filter/column_filter.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+		},
+	}
+
+	for _, arg := range args {
+		if err := p.parse(arg, true); err != nil {
+			return nil, err
+		}
+	}
+
+	reversing(p.rules)
```
Comment (lichunzhu) [inline pkg/table-filter/column_filter.go]:
```suggestion
	reverse(p.rules)
```

**Comment by 0xPoe** [comment]
/unhold

RFC is being merged.

@kennytm @gozssky Could you please take a look? Thanks!

**Review by lichunzhu — APPROVED** [review]
LGTM

File: pkg/table-filter/column_filter.go [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+	return columnFilter(p.rules), nil
+}
+
+// MatchColumn checks if a column can be processed after applying the columnFilter `f`.
+// Column is not case-sensitive on any platform, nor are column aliases.
```
Comment (lichunzhu) [inline pkg/table-filter/column_filter.go:51]: How about putting the reference doc here.
https://dev.mysql.com/doc/refman/5.7/en/identifier-case-sensitivity.html

File: pkg/table-filter/column_filter.go (hunk 5 above)
Comment (0xPoe) [reply pkg/table-filter/column_filter.go]: Thanks!

File: pkg/table-filter/column_filter.go (hunk 6 above)
Comment (0xPoe) [reply pkg/table-filter/column_filter.go:51]: Good idea! Added. Thanks for your review!

**Review by sleepymole — APPROVED** [review]

**Pull request merged**